node/src/components/block_proposer.rs

```diff
@@ -332,7 +336,26 @@ impl BlockProposerReady {
         if self.sets.finalized_deploys.contains_key(&hash) {
             info!(%hash, "deploy rejected from the buffer");
         } else {
-            self.sets.pending.insert(hash, deploy_or_transfer);
+            let gas_price = deploy_or_transfer.header().gas_price();
+            let payment_amount_gas = if deploy_or_transfer.is_transfer() {
+                Gas::from(self.wasmless_transfer_cost)
+            } else {
+                match Gas::from_motes(deploy_or_transfer.payment_amount_motes(), gas_price) {
+                    Some(value) => value,
+                    None => {
+                        info!(
+                            "could not convert motes to gas for wasm deploy {} at gas price {}",
+                            hash, gas_price
+                        );
+                        return;
+                    }
+                }
+            };
```

Comment:
So `wasmless_transfer_cost` is currently hard-coded in _motes_, not _gas_? I guess we'll change that to be in gas, in the future, and multiply that by `gas_price` as well?

But for now, if we pay the same total amount for every wasmless transfer, shouldn't the gas price also be the same for all of them?

Reply:
So the chainspec value for `wasmless_transfer_cost` is in fact gas, however the `payment_amount` here has to be stored in `Motes` by the `DeployType` such that it can then be scaled/converted from `Motes` by the specified `gas_price`.

Reply:
But which gas price is used to convert the gas to motes then?
It should be _this_ transfer's `gas_price`, shouldn't it?

Reply:
@afck, if payment_amount did what it's supposed to do (store a total payment amount...in motes), you could just sort it by that, instead of the _implied_ gas price. The problem seems to be that payment_amount attached to wasmless transfer deploys is in _gas_?

Reply:
It looks like wasmless transfers have an attached `gas_price` and a constant transfer cost (originally in gas), and the payment amount is their product.
The weird thing is that the `wasmless_transfer_cost` here is in motes, not gas, because it has been multiplied earlier (elsewhere in the code) by the formerly constant conversion rate?

Reply:
See comment on line 344 of block_proposer.rs